### PR TITLE
Align the podman ps --filter behavior with docker

### LINKED ISSUE #8344: status filters for podman ps and podman pod ps should be inclusive
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

I want to filter the output for `podman ps`  to only contain created and exited containers for example.

Since filters are exclusive I cannot do this.

**Steps to reproduce the issue:**

1. `podman create alpine`

2. `podman ps --filter status=created,status=exited`

**Describe the results you received:**
empty output

**Describe the results you expected:**
show all created and exited containers

**Output of `podman version`:**

```
$ bin/podman version
Version:      2.2.0-dev
API Version:  2.0.0
Go Version:   go1.14.10
Git Commit:   4eb9c28433f8aa1827f844881efb1ae1d15b0860
Built:        Sat Nov 14 16:50:57 2020
OS/Arch:      linux/amd64
```

### PR DESCRIPTION
**Pull request by Luap99**
All of our filters worked exclusive resulting in `--filter status=created --filter status=exited` to return nothing.

In docker filters with the same key work inclusive with the only exception being `label` which is exclusive. Filters with different keys always work exclusive.

This PR aims to match the docker behavior with podman.

Fixes #8344

### CODE REVIEW

**Comment by Luap99** [comment]
I will do the same for `podman pod ps` next but this can go in a different PR

**Comment by rhatdan** [comment]
LGTM
@containers/podman-maintainers PTAL

File: libpod/filters/containers.go [hunk 1]
```diff
@@ -11,101 +10,133 @@ import (
...
 		}, nil
 	case "before":
-		ctr, err := r.LookupContainer(filterValue)
-		if err != nil {
-			return nil, errors.Errorf("unable to find container by name or id of %s", filterValue)
+		var createTime time.Time
+		for _, filterValue := range filterValues {
+			ctr, err := r.LookupContainer(filterValue)
+			if err != nil {
+				return nil, errors.Errorf("unable to find container by name or id of %s", filterValue)
```
Comment (saschagrunert) [inline libpod/filters/containers.go:117]:
```suggestion
				return nil, errors.Wrapf(err, "unable to find container by name or id of %s", filterValue)
```

File: libpod/filters/containers.go [hunk 2]
```diff
@@ -11,101 +10,133 @@ import (
...
 		}, nil
 	case "since":
-		ctr, err := r.LookupContainer(filterValue)
-		if err != nil {
-			return nil, errors.Errorf("unable to find container by name or id of %s", filterValue)
+		var createTime time.Time
+		for _, filterValue := range filterValues {
+			ctr, err := r.LookupContainer(filterValue)
+			if err != nil {
+				return nil, errors.Errorf("unable to find container by name or id of %s", filterValue)
```
Comment (saschagrunert) [inline libpod/filters/containers.go:133]:
```suggestion
				return nil, errors.Wrapf(err, "unable to find container by name or id of %s", filterValue)
```

**Review by saschagrunert — COMMENTED** [review]
Just two nits, otherwise LGTM

File: cmd/podman/containers/ps.go [hunk 3]
```diff
@@ -98,14 +98,6 @@ func checkFlags(c *cobra.Command) error {
 	if listOpts.Last >= 0 && listOpts.Latest {
 		return errors.Errorf("last and latest are mutually exclusive")
 	}
-	// Filter on status forces all
```
Comment (mheon) [inline cmd/podman/containers/ps.go:101]: It's not immediately clear what's replacing this - do we still ensure that all is set?

Comment (Luap99) [reply cmd/podman/containers/ps.go:101]: Yes it is checked later in the code path again.
https://github.com/containers/podman/blob/a6f7afd41729edbe0d8cbf21bab5b475c6a13f9e/pkg/ps/ps.go#L40-L44

File: libpod/filters/containers.go (hunk 1 above)
Comment (Luap99) [reply libpod/filters/containers.go:117]: Actually I can just return `err` here. The error from `LookupContainer` is `no container with name or ID %s found: no such container`. Wrapping this wouldn't provide more information

File: cmd/podman/containers/ps.go (hunk 3 above)
Comment (mheon) [reply cmd/podman/containers/ps.go:101]: Wow. Wonder how that slipped in. OK, thanks!

**Comment by rhatdan** [comment]
Thanks @Luap99 
/lgtm

**Pull request merged**